src/Avalonia.Controls/ContextMenu.cs

```diff
@@ -197,18 +224,51 @@ public bool WindowManagerAddShadowHint
         }
 
         /// <summary>
-        /// Occurs when the value of the
-        /// <see cref="P:Avalonia.Controls.ContextMenu.IsOpen" />
-        /// property is changing from false to true.
+        /// Defines the <see cref="Opened"/> event.
+        /// </summary>
+        public static readonly RoutedEvent<ContextMenuEventArgs> OpenedEvent =
+            RoutedEvent.Register<ContextMenu, ContextMenuEventArgs>(nameof(Opened), RoutingStrategies.Bubble);
+
+        /// <summary>
+        /// Occurs when a <see cref="ContextMenu"/> is opened.
+        /// </summary>
+        public event EventHandler<ContextMenuEventArgs> Opened
```

Comment:
Do we really need to have this breaking change from ContextMenuOpening to Opened? Similarly to other events in this class.
I though we keep ContextMenu in Avalonia only because of historical reasons and compatibility with WPF?

Reply:
That´s a good point. There is no reason to keep ContextMenu in my opinion if there is a breaking change from WPF anyway. It´s a good time to switch to MenuFlyout´s.

Reply:
I think we should keep `ContextMenu`, and migrate it to use native menus on relevant platforms as part of https://github.com/AvaloniaUI/Avalonia/issues/3855. With that in mind, unifying the events would be a good idea.

Reply:
@grokys ContextFlyout API is part of future native menu API. 
It was discussed to be able tool write somethinf similar to:
```<Control.ContextFlyout>
   <NativeMenuFlyout>
       <NativeMenuItem />
   </NativeMenuFlyout>
</Context.ContextFlyout>
```
Or
```<Control.ContextFlyout>
   <MenuFlyout ShouldUseNativeMenu="True" >
       <MenuItem />
   </MenuFlyout>
</Context.ContextFlyout>
```

In any way existing Flyout API is more flexible comparing with ContextMenu. 
Also it's not limited to right-click, and can be involed with public API on any control by any event.